tools/assetcompressor/main.cpp

```diff
@@ -0,0 +1,134 @@
+*/
+
+#include <Kuesa/private/gltf2parser_p.h>
+#include <Kuesa/private/gltf2exporter_p.h>
+#include <Kuesa/SceneEntity>
+
+#include <QCoreApplication>
+#include <QDir>
+#include <QCommandLineParser>
+#include <QElapsedTimer>
+
+int main(int argc, char *argv[])
+{
+    using namespace Kuesa;
+    using namespace Kuesa::GLTF2Import;
+
+    QCoreApplication app(argc, argv);
+    app.setApplicationName(QStringLiteral("Kuesa Asset Compressor"));
+    app.setOrganizationDomain(QStringLiteral("kdab.com"));
+    app.setOrganizationName(QStringLiteral("KDAB"));
+    app.setApplicationVersion(QStringLiteral("1.0"));
+
+    QCommandLineParser cmdline;
+    cmdline.addHelpOption();
+    cmdline.addVersionOption();
+    cmdline.addPositionalArgument("input", QObject::tr("glTF2 file to compress"));
+    cmdline.addPositionalArgument("output", QObject::tr("Target directory"));
+
+    QCommandLineOption compRatio({ "r", "ratio" }, QObject::tr("Compression ratio, between 1 and 16"), QString(), "7");
+    cmdline.addOption(compRatio);
+
+    cmdline.process(app);
+
+    const auto &args = cmdline.positionalArguments();
+    if (args.size() != 2) {
+        qCritical("Invalid arguments specified.\n%s", cmdline.helpText().toLatin1().constData());
+        return 1;
+    }
+
+    int ratio = cmdline.value(compRatio).toInt();
+    if (ratio < 1 || ratio > 16) {
+        qCritical("Invalid compression ratio.");
+        return 1;
+    }
+
+    QString asset = args[0];
+    if (!QFileInfo::exists(asset)) {
+        qCritical("Could not find input file.");
+        return 1;
+    }
+
+    QDir sourceDir = QFileInfo(asset).dir();
+    QDir targetDir = args[1];
+    if (!targetDir.exists()) {
+        if (!QDir{}.mkpath(args[1])) {
+            qCritical("Could not create target directory.");
+            return 1;
+        }
+    }
+
+    // Parsing
+    SceneEntity scene;
+    GLTF2Context ctx;
+
+    GLTF2Parser parser(&scene);
+    parser.setContext(&ctx);
+
+    auto res = parser.parse(asset);
+    if (!res) {
+        qCritical("Could not parse glTF2 file.");
+        return 1;
+    }
+
+    // Compression
+    GLTF2ExportConfiguration configuration;
+    configuration.setMeshCompressionEnabled(true);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::Position, ratio);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::Normal, ratio);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::Color, ratio);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::TextureCoordinate, ratio);
+    configuration.setAttributeQuantizationLevel(GLTF2ExportConfiguration::Generic, ratio);
+
+    GLTF2Exporter exporter;
+    exporter.setContext(&ctx);
+    exporter.setScene(&scene);
+    exporter.setConfiguration(configuration);
+
+    QElapsedTimer timer;
+    timer.start();
+
+    auto new_asset = exporter.saveInFolder(sourceDir, targetDir);
```

Comment:
Do we even care about new_asset here? I don't see it being used below

Reply:
Indeed.

Reply:
that was a bug :)